src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -59,6 +61,30 @@ public override void Initialize (AnalysisContext context)
 				var invocationOperation = (IInvocationOperation) context.Operation;
 				ProcessInvocationOperation (context, invocationOperation);
 			}, OperationKind.Invocation);
+			context.RegisterSyntaxNodeAction (context => {
+				ProcessSyntaxNode (context);
+			}, SyntaxKind.GenericName);
+		}
+
+		private void ProcessSyntaxNode (SyntaxNodeAnalysisContext context)
+		{
+			if (context.SemanticModel.GetTypeInfo (context.Node).ConvertedType is not INamedTypeSymbol type)
+				return;
+
+			// The same INamedTypeSymbol can be called back resulting in duplicate warnings
+			// One way to filter is via the Node's Parent
+			if (context.Node.Parent is not ArgumentSyntax)
+				return;
+
+			var typeParams = type!.TypeParameters;
+			var typeArgs = type.TypeArguments;
+
+			for (int i = 0; i < typeParams.Length; i++) {
+				var sourceValue = GetTypeValueNodeFromGenericArgument (typeArgs[i]);
```

Comment:
What if the number of type arguments is less than the number of type parameters or vice versa?

Reply:
Added an assert